webnn/resources/test_data/gather.json

```diff
@@ -0,0 +1,1268 @@
+          43.84803771972656,
+          43.84803771972656,
+          48.81806945800781,
+          51.79948425292969,
+          51.79948425292969,
+          51.79948425292969,
+          41.94132614135742,
+          -1.1303654909133911,
+          -1.1303654909133911,
+          -1.1303654909133911,
+          -50.42131042480469,
+          90.2870101928711,
+          90.2870101928711,
+          90.2870101928711,
+          55.620765686035156,
+          44.92119598388672,
+          44.92119598388672,
+          44.92119598388672,
+          56.828636169433594,
+          10.829925537109375,
+          10.829925537109375,
+          10.829925537109375,
+          -19.693084716796875,
+          -37.696800231933594,
+          -37.696800231933594,
+          -37.696800231933594,
+          43.11057662963867,
+          0.9129875898361206,
+          0.9129875898361206,
+          0.9129875898361206,
+          -7.699817180633545,
+          25.76774024963379,
+          25.76774024963379,
+          25.76774024963379,
+          73.60064697265625
+        ],
+        "type": "float32"
+      }
+    },
+    {
+      "name": "gather float32 4D tensor and 2D indices explict options.axis=0",
+      "inputs": {
+        "input": {
+          "shape": [8, 1, 1, 3],
+          "data": [
+            -66.05901336669922,
+            -68.9197006225586,
+            -77.02045440673828,
+            -26.158037185668945,
+            89.0337142944336,
+            -45.89653396606445,
+            43.84803771972656,
+            48.81806945800781,
+            51.79948425292969,
+            41.94132614135742,
+            -1.1303654909133911,
+            -50.42131042480469,
+            90.2870101928711,
+            55.620765686035156,
+            44.92119598388672,
+            56.828636169433594,
+            10.829925537109375,
+            -19.693084716796875,
+            -37.696800231933594,
+            43.11057662963867,
+            0.9129875898361206,
+            -7.699817180633545,
+            25.76774024963379,
+            73.60064697265625
+          ],
+          "type": "float32"
+        },
+        "indices": {
+          "shape": [2, 2],
+          "data": [
+            0,
+            0,
+            7,
+            4
+          ],
+          "type": "int64"
+        }
+      },
+      "options": {
+        "axis": 0
+      },
+      "expected": {
+        "name": "output",
+        "shape": [2, 2, 1, 1, 3],
+        "data": [
+          -66.05901336669922,
+          -68.9197006225586,
+          -77.02045440673828,
+          -66.05901336669922,
+          -68.9197006225586,
+          -77.02045440673828,
+          -7.699817180633545,
+          25.76774024963379,
+          73.60064697265625,
+          90.2870101928711,
+          55.620765686035156,
+          44.92119598388672
+        ],
+        "type": "float32"
+      }
+    },
+    {
+      "name": "gather float32 5D tensor and 0D scalar indices options.axis=4",
```

Comment:
There's an issue https://bugs.chromium.org/p/chromium/issues/detail?id=1513835, @shiyi9801 has found the root cause and would submit a fixing CL.

Reply:
Thanks for opening the issue! I uploaded a [CL](https://chromium-review.googlesource.com/c/chromium/src/+/5147094) to fix it. 

After some tests I found that, it seems when the indices tensor is a 0-D scalar, the output dimensions should add a trailing one value instead a leading one, so that the input rank matches the output rank.
Though in the [documentation](https://learn.microsoft.com/en-us/windows/win32/api/directml/ns-directml-dml_gather_operator_desc) it says "The dimensions are *right-aligned* such that any leading 1 values from the input sizes are cropped which would otherwise overflow the output DimensionCount."🤔

Reply:
The fixing CL is merged. Please run the tests again to see if this issue is resolved, thanks!

Reply:
Verified! This issue has been fixed. Thanks @shiyi9801 !